src/blocked.c

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright Redis Ltd. 2023 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "redisraft.h"
+
+#include <math.h>
+#include <string.h>
+#include <strings.h>
+
+/* A list of all current blocked commands */
+
+BlockedCommand *allocBlockedCommand(const char *cmd_name, raft_index_t idx, raft_session_t session, const char *data, size_t data_len, RaftReq *req, RedisModuleCallReply *reply)
+{
+    BlockedCommand *bc = RedisModule_Calloc(1, sizeof(BlockedCommand));
+    bc->command = RedisModule_Strdup(cmd_name);
+    bc->idx = idx;
+    bc->session = session;
+    bc->data = RedisModule_Alloc(data_len);
+    memcpy(bc->data, data, data_len);
+    bc->data_len = data_len;
+    bc->req = req;
+    bc->reply = reply;
+
+    return bc;
+}
+
+void addBlockedCommand(BlockedCommand *bc)
+{
+    sc_list_init(&bc->blocked_list);
+    sc_list_add_tail(&redis_raft.blocked_command_list, &bc->blocked_list);
+    RedisModule_DictSetC(redis_raft.blocked_command_dict, &bc->idx, sizeof(bc->idx), bc);
+}
+
+void deleteBlockedCommand(raft_index_t idx)
+{
+    BlockedCommand *blocked = NULL;
```

Comment:
```suggestion
    BlockedCommand *blocked = NULL;

```
Please leave one empty line after variable declarations

Reply:
done